lib/ClangImporter/ImportMacro.cpp

```diff
@@ -371,6 +373,94 @@ getIntegerConstantForMacroToken(ClangImporter::Implementation &impl,
   return std::nullopt;
 }
 
+namespace {
+ValueDecl *importDeclAlias(ClangImporter::Implementation &clang,
+                           swift::DeclContext *DC, const clang::ValueDecl *D,
+                           Identifier alias) {
+  // Ignore self-referential macros.
+  if (D->getName() == alias.str())
```

Comment:
What I'm saying is, here we should check what the imported name will be, rather than what the clang name is. There's some complex logic for the complete set of name changes that can occur during importing, but checking for `SwiftNameAttr` seems feasible, and is the most likely one to cause issues. Of course, if we (somehow) have access to the imported Swift decl here, we should just check the already imported name, but I'm assuming we don't?